src/sentry/spans/consumers/detect_performance_issues/factory.py

```diff
@@ -30,6 +30,8 @@ def process_message(message: Message[KafkaPayload]):
     segment = _deserialize_segment(value)
     metrics.incr("detect_performance_issues.spans.count", len(segment["spans"]))
 
+    assert len(segment["spans"]) > 0
```

Comment:
we aren't producing messages if they don't have spans, this is just a sanity check